File: client/common/bet_formatter.go
package common

import (
	"fmt"
	"strconv"
	"strings"
)

// FormatBetMessage formats a bet as a string separated by '|'.
func FormatBetMessage(bet *Bet) string {
	return fmt.Sprintf("%s|%s|%s|%s|%d|%d",
		bet.firstName,
		bet.lastName,
		bet.documentNumber,
		bet.dob,
		bet.number,
		bet.agencyId,
	)
}

func ParseBetMessage(msg string) (*Bet, error) {
	fields := strings.Split(msg, "|")
	if len(fields) != 6 {
		return nil, fmt.Errorf("invalid bet message format")
	}

	number, err := strconv.Atoi(fields[4])
	if err != nil {
		return nil, fmt.Errorf("invalid number field: %w", err)
	}

	agencyId, err := strconv.Atoi(fields[5])
	if err != nil {
		return nil, fmt.Errorf("invalid agency id field: %w", err)
	}

	return &Bet{
		firstName:      fields[0],
		lastName:       fields[1],
		documentNumber: fields[2],
		dob:            fields[3],
		number:         number,
		agencyId:       agencyId,
	}, nil
}

File: server/common/bet_formatter.py
from common.utils import Bet


def format_bet_message(bet):
    return f"{bet.first_name}|{bet.last_name}|{bet.document_number}|{bet.dob}|{bet.number}|{bet.agency}"


def parse_bet_message(message):
    fields = message.split("|")
    if len(fields) != 6:
        print(message)
        raise ValueError("Invalid bet message format")
    return Bet(
        agency=fields[5],
        first_name=fields[0],
        last_name=fields[1],
        document=fields[2],
        birthdate=fields[3],
        number=fields[4],
    )

def parse_str_to_bets(raw_msg: str) -> list[Bet]:
    lines = raw_msg.strip().split('\n')
    bets = []
    for line in lines:
        bet = _parse_str_to_bet(line)
        bets.append(bet)
    return bets